packages/smart-wallet/src/offerWatcher.js

```diff
@@ -274,6 +284,8 @@ export const prepareOfferWatcher = baggage => {
           } else {
             facets.helper.handleError(reason);
           }
+          // throw so the vow watcher propagates the rejection
+          throw reason;
```

Comment:
Curious how rejections were propagated before - this change seems unrelated to the rest

Reply:
the function returned the promise, but now that's wrapped in vow watchers